src/main/java/com/algolia/search/AnalyticsConfig.java

```diff
@@ -1,44 +1,57 @@
 package com.algolia.search;
 
 import com.algolia.search.models.common.CallType;
-import java.util.Collections;
-import java.util.EnumSet;
-import java.util.List;
+import java.util.*;
 import javax.annotation.Nonnull;
 
-@SuppressWarnings({"WeakerAccess", "unused"})
+@SuppressWarnings({"unused"})
 public final class AnalyticsConfig extends ConfigBase {
 
-  /**
-   * Creates an {@link AnalyticsConfig} with the default host
-   *
-   * @param applicationID The Application ID
-   * @param apiKey The API Key
-   */
-  public AnalyticsConfig(String applicationID, String apiKey) {
-    super(applicationID, apiKey);
+  public static class Builder extends ConfigBase.Builder<Builder> {
 
-    List<StatefulHost> hosts =
-        Collections.singletonList(
-            new StatefulHost("analytics.algolia.com", EnumSet.of(CallType.READ, CallType.WRITE)));
+    /**
+     * Creates an {@link AnalyticsConfig} with the default host
+     *
+     * @param applicationID The Application ID
+     * @param apiKey The API Key
+     */
+    public Builder(@Nonnull String applicationID, @Nonnull String apiKey) {
+      super(applicationID, apiKey);
 
-    this.setDefaultHosts(hosts);
+      List<StatefulHost> hosts =
+          Collections.singletonList(
+              new StatefulHost("analytics.algolia.com", EnumSet.of(CallType.READ, CallType.WRITE)));
+
+      this.setDefaultHosts(hosts);
+    }
+
+    /**
+     * Creates a {@link AnalyticsConfig} for the search client with custom {@link StatefulHost}.
+     * Warning: Defaults hosts are not set when setting custom {@link StatefulHost}.
+     *
+     * @param applicationID The ApplicationID
+     * @param apiKey The API Key
+     * @param customHosts List of custom hosts
+     */
+    public Builder(
+        @Nonnull String applicationID,
+        @Nonnull String apiKey,
+        @Nonnull List<StatefulHost> customHosts) {
+      super(applicationID, apiKey);
+      this.setCustomHosts(customHosts);
+    }
+
+    @Override
+    public Builder getThis() {
+      return this;
+    }
+
+    public AnalyticsConfig build() {
+      return new AnalyticsConfig(this);
+    }
   }
 
-  /**
-   * Creates a {@link AnalyticsConfig} for the search client with custom {@link StatefulHost}.
-   * Warning: Defaults hosts are not set when setting custom {@link StatefulHost}.
-   *
-   * @param applicationID The ApplicationID
-   * @param apiKey The API Key
-   * @param customHosts List of custom hosts
-   */
-  public AnalyticsConfig(
-      @Nonnull String applicationID,
-      @Nonnull String apiKey,
-      @Nonnull List<StatefulHost> customHosts) {
-    super(applicationID, apiKey);
-
-    this.setCustomHosts(customHosts);
+  private AnalyticsConfig(Builder builder) {
+    super(builder);
```

Comment:
The pattern I've seen the most: keeping AnalyticsConfig constructor as before (= the constructor with all parameters) but _private_ - or internal, and have the builder invokes it in its "build" method. This way the coupling between the two is loose (which is desirable from a maintenance/evolutivity point of view)

Reply:
Yes! I've seen both patterns, I found the current one easier to implement with the current logic of the hosts. But if we decide to drop the Builder constructor with the hosts. i.e having only one with AppID and APIKey it could be nice to have this pattern. (see above for the constructors) :)